src/utils/git.ts

```diff
@@ -21,15 +51,25 @@ export function getGitRepoAndBranch() {
       normalizedRepositoryName = repositoryName.endsWith(".git")
         ? repositoryName?.slice(0, -DOT_GIT_SUFFIX_LENGTH)
         : repositoryName;
+      const defaultBranch = await getDefaultBranch(repository.rootUri.fsPath);
+      if (!defaultBranch) {
+        showCannotGetDefaultBranchMessage();
+      } else {
+        isDefaultBranch = currentBranch === defaultBranch;
+      }
     }
   }
 
-  return { repository: normalizedRepositoryName, currentBranch };
+  return {
+    repository: normalizedRepositoryName,
+    currentBranch,
+    isDefaultBranch,
```

Comment:
Why is this part of this function? 🤔

Reply:
why not 🤷🏻‍♂️

Reply:
If its always used when this func is used then its fine imo ( but if its sometimes needed and sometimes it doesn't id remove it to its own func, because it doesn't really depend on anything here )